lib/Model/LinkedDataSignature.php

```diff
@@ -237,19 +260,23 @@ public function verify(): bool {
 
 		$header = [
 			'@context' => 'https://w3id.org/identity/v1',
+			'nonce'    => $this->getNonce(),
 			'creator'  => $this->getCreator(),
 			'created'  => $this->getCreated()
 		];
 
-		$hash = $this->hashedCanonicalize($header) . $this->hashedCanonicalize($this->getObject());
-		$signed = base64_decode($this->getSignatureValue());
+		$hashHeader = $this->hashedCanonicalize($header, true);
+		$hashObject = $this->hashedCanonicalize($this->getObject());
 
 		$algo = OPENSSL_ALGO_SHA256;
 		if ($this->getType() === 'RsaSignature2017') {
 			$algo = OPENSSL_ALGO_SHA256;
 		}
 
-		if (openssl_verify($hash, $signed, $this->getPublicKey(), $algo) === 1) {
+		if (openssl_verify(
+				$hashHeader . $hashObject, base64_decode($this->getSignatureValue()),
```

Comment:
Maybe we should catch a failure in base64_decode as well

Reply:
done